gapii/cc/vulkan_mid_execution.cpp

```diff
@@ -153,1690 +156,1780 @@ class TemporaryRenderPass {
+      if (a->mqueueCount < num_queues) {
+        a->mqueueCount = num_queues;
       }
+      queue_priorities[a - queue_create_infos.begin()].resize(num_queues);
+      queue_priorities[a - queue_create_infos.begin()][queue_index] =
+          queue_priority;
     }
 
-    // Recreate CreateImages
-    {
-      VkImageCreateInfo image_create_info{};
-      image_create_info.msType =
-          VkStructureType::VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO;
-      for (auto& image : Images) {
-        if (image.second->mIsSwapchainImage) {
-          // Don't recreate the swapchain images
-          continue;
-        }
-        ImageInfo& info = image.second->mInfo;
-        std::vector<uint32_t> queues(info.mQueueFamilyIndices.size());
-        for (size_t i = 0; i < info.mQueueFamilyIndices.size(); ++i) {
-          queues[i] = info.mQueueFamilyIndices[i];
-        }
-        image_create_info.mflags = info.mFlags;
-        image_create_info.mimageType = info.mImageType;
-        image_create_info.mformat = info.mFormat;
-        image_create_info.mextent = info.mExtent;
-        image_create_info.mmipLevels = info.mMipLevels;
-        image_create_info.marrayLayers = info.mArrayLayers;
-        image_create_info.msamples = info.mSamples;
-        image_create_info.mtiling = info.mTiling;
-        image_create_info.musage = info.mUsage;
-        image_create_info.msharingMode = info.mSharingMode;
-        image_create_info.mqueueFamilyIndexCount = queues.size();
-        image_create_info.mpQueueFamilyIndices = queues.data();
-        image_create_info.minitialLayout =
-            VkImageLayout::VK_IMAGE_LAYOUT_UNDEFINED;
-
-        // Empty NV dedicated allocation struct
-        VkDedicatedAllocationImageCreateInfoNV dedicated_allocation_create_info{
-            VkStructureType::
-                VK_STRUCTURE_TYPE_DEDICATED_ALLOCATION_IMAGE_CREATE_INFO_NV,  // sType
-            nullptr,          // pNext
-            VkBool32(false),  // dedicatedAllocation
-        };
-        // If the buffer is created with Dedicated Allocation NV extension,
-        // we need to populate the pNext pointer here.
-        if (info.mDedicatedAllocationNV) {
-          dedicated_allocation_create_info.mdedicatedAllocation =
-              info.mDedicatedAllocationNV->mDedicatedAllocation;
-          image_create_info.mpNext =
-                  &dedicated_allocation_create_info;
+    for (size_t i = 0; i < queue_create_infos.size(); ++i) {
+      auto& v = queue_create_infos[i];
+      v.mpQueuePriorities = queue_priorities[i].data();
+    }
+    create_info.mqueueCreateInfoCount = queue_create_infos.size();
+    create_info.mpQueueCreateInfos = queue_create_infos.data();
+    create_info.mpEnabledFeatures = &device.second->mEnabledFeatures;
+
+    VkDevice d = device.second->mVulkanHandle;
+    VkPhysicalDevice pd = device.second->mPhysicalDevice;
+    RecreateDevice(observer, pd, &create_info, &d);
+    recreateDebugInfo(
```

Comment:
For each recreate vulkan handle, createDebugInfo() is called, although I don't think debug info can be attached to VkInstance, VkPhysicalDevice, VkSurface.